### PR TITLE
Fix localhost usage

### PR DESCRIPTION
**Pull request by robertdale**
Fixes Issue #792

Signed-off-by: Robert Dale <robdale@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-cassandra/src/test/java/org/janusgraph/graphdb/thrift/ThriftConnectionTest.java [hunk 1]
```diff
@@ -46,11 +46,12 @@ public void setUp() throws Exception {
             Schema.instance.load(KSMetaData.newKeyspace("janusgraph", "SimpleStrategy", ImmutableMap.of("replication_factor", "1"), true));
             
             log.debug("Starting ThriftServer for connection drop on port [9099]");
-            ThriftServer server = new ThriftServer(InetAddress.getLocalHost(), 9099, 9098);
+            InetAddress localhost = InetAddress.getLoopbackAddress();
```
Comment (amcp) [inline janusgraph-cassandra/src/test/java/org/janusgraph/graphdb/thrift/ThriftConnectionTest.java]: final

File: janusgraph-cassandra/src/test/java/org/janusgraph/graphdb/thrift/ThriftConnectionTest.java [hunk 2]
```diff
@@ -46,11 +46,12 @@ public void setUp() throws Exception {
...
+            ThriftServer server = new ThriftServer(localhost, 9099, 9098);
```
Comment (amcp) [inline janusgraph-cassandra/src/test/java/org/janusgraph/graphdb/thrift/ThriftConnectionTest.java]: make final, and also reuse the port numbers by extracting some constants

File: janusgraph-core/src/main/java/org/janusgraph/util/system/NetworkUtil.java [hunk 3]
```diff
@@ -20,17 +20,7 @@
...
-        // Aside from that, getLoopbackAddress() seems preferable to
-        // InetAddress.getByName("localhost") since the former doesn't seem to
-        // require the local resolver to be sane.
-        //return InetAddress.getLoopbackAddress().getHostAddress();
-        try {
-            return InetAddress.getByName("localhost").getHostAddress();
-        } catch (UnknownHostException e) {
-            throw new RuntimeException(e);
-        }
+        return InetAddress.getLoopbackAddress().getHostAddress();
```
Comment (amcp) [inline janusgraph-core/src/main/java/org/janusgraph/util/system/NetworkUtil.java]: Does getLoopbackAddress never throw UnknownHostException?

Comment (robertdale) [reply janusgraph-core/src/main/java/org/janusgraph/util/system/NetworkUtil.java]: It does not throw any exceptions declared or appear to otherwise.

**Comment by robertdale** [comment]
@amcp Fixed.

**Pull request merged**